core/src/services/fs/writer.rs

```diff
@@ -58,9 +62,23 @@ impl oio::Write for FsWriter<tokio::fs::File> {
         self.f.sync_all().await.map_err(parse_io_error)?;
 
         if let Some(tmp_path) = &self.tmp_path {
-            tokio::fs::rename(tmp_path, &self.target_path)
-                .await
-                .map_err(parse_io_error)?;
+            if self.op.append() {
```

Comment:
I believe it is not possible to perform this action. If users attempt to add 1MiB of data to a 10GiB file, they will be surprised by the outcome. 

If the append function is enabled, we need to check if the file exists. If the file does not exist, we should create a new file, append and then rename it (like what we do for `write`). Otherwise, we can directly append to the existing file.

Reply:
> I believe it is not possible to perform this action. If users attempt to add 1MiB of data to a 10GiB file, they will be surprised by the outcome.

I understand what you're saying. This will result in 2x the amount of data being written.🤔 
But for atomic, it will be inevitable if the file exists. 😢

Reply:
> But for atomic, it will be inevitable if the file exists. 😢

We can append to existing file if `append` is true.